mkosi/config.py

```diff
@@ -1697,23 +1710,51 @@ def parse(self, argv: Optional[Sequence[str]] = None) -> tuple[MkosiArgs, MkosiC
         if args.directory:
             os.chdir(args.directory)
 
-        self.parse_config(Path("."), namespace)
+        if Path("mkosi.profiles").exists():
+            for p in sorted(Path("mkosi.profiles").iterdir()):
+                name = p.name.lstrip(string.digits + "-").removesuffix(".conf")
+                if not name:
+                    die(f"{p} is not a valid profile name")
+                if args.profiles and name not in args.profiles:
+                    continue
 
-        for s in self.SETTINGS:
-            if s.dest in namespace:
-                continue
+                cp = copy.deepcopy(namespace)
 
-            if s.default_factory:
-                default = s.default_factory(namespace)
-            elif s.default is None:
-                default = s.parse(s.dest, None, namespace)
-            else:
-                default = s.default
+                with chdir(p if p.is_dir() else Path.cwd()):
+                    self.parse_config(p if p.is_file() else Path("."), cp)
+
+                self.parse_config(Path("."), cp)
```

Comment:
I think this section could do with a few comments. The idea is that we pop into the profile dir, parse the profile config and then pop back and parse our global config there?